lib/workers/repository/config-migration/branch/rebase.ts

```diff
@@ -55,3 +60,10 @@ export async function rebaseMigrationBranch(
     platformCommit: !!config.platformCommit,
   });
 }
+
+function stripWhitespaces(str: string | null): string | null {
```

Comment:
This function is called stripWhitespaces but what it does is stringify. Is there a trick here? And why's white space an issue now?

Reply:
This function effectively removes any JSON formatting. 
As the repo's config might be prettier formatted and we don't run prettier on the runtime generated config till later in code, therefore we must remove any existing formatting in order to compare the raw configs.

This function is only intended for JSON strings, therefore I might rename it `JsonStripWhitespaces`.

The trick being used
> `JSON.stringify(value, replacer, space)`:
> 
> If space is anything other than a string or number (can be either a primitive or a wrapper object) — for example, is [null](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null) or not provided — no white space is used.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#parameters